core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,805 @@ def update_question_suggestion(
+    )
+    return [
+        _create_question_contribution_stats_from_model(model)
+        for model in question_contribution_stats_models
+    ]
+
+
+def _get_all_question_review_stats(
+    user_id: str) -> List[suggestion_registry.QuestionReviewStats]:
+    """Gets all QuestionReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionReviewStats). QuestionReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    question_review_stats_models = (
+        suggestion_models.QuestionReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_review_stats_from_model(model)
+        for model in question_review_stats_models
+    ]
+
+
+def get_all_contributor_stats(
+    user_id: str) -> suggestion_registry.ContributorStatsSummary:
+    """Gets ContributorStatsSummary corresponding to the supplied user.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        ContributorStatsSummary. ContributorStatsSummary domain objects
+        corresponding to the supplied user.
+    """
+    translation_contribution_stats = get_all_translation_contribution_stats(
+        user_id)
+    translation_review_stats = _get_all_translation_review_stats(user_id)
+    question_contribution_stats = _get_all_question_contribution_stats(user_id)
+    question_review_stats = _get_all_question_review_stats(user_id)
+
+    return suggestion_registry.ContributorStatsSummary(
+        user_id,
+        translation_contribution_stats,
+        question_contribution_stats,
+        translation_review_stats,
+        question_review_stats)
+
+
+def _update_translation_contribution_stats_models(
+    translation_contribution_stats: List[
+        suggestion_registry.TranslationContributionStats]) -> None:
+    """Updates TranslationContributionStatsModel models for given translation
+    contribution stats.
+
+    Args:
+        translation_contribution_stats: list(TranslationContributionStats).
+            A list of TranslationContributionStats domain objects.
+    """
+    stats_ids = []
+    stats_dict = {}
+
+    for stat in translation_contribution_stats:
+        # Ruling out the possibility of None for mypy type checking.
+        assert stat is not None
+        stat_id = suggestion_models.TranslationContributionStatsModel.build_id(
+            stat.language_code, stat.contributor_user_id, stat.topic_id) # type: ignore[arg-type]
+        stats_ids.append(stat_id)
+        stats_dict[stat_id] = stat
+
+    stats_models_to_update = (
+        suggestion_models.TranslationContributionStatsModel.get_multi(
+            stats_ids))
+
+    for stats_model in stats_models_to_update:
+
+        # Ruling out the possibility of None for mypy type checking.
+        assert stats_model is not None
+        stat = stats_dict[stats_model.id]
```

Comment:
You can use `zip` here and put the `stat = stats_dict[stats_model.id]` into the for loop header.

ditto in all the other similar functions

Reply:
I think we will have to use zip like this right?

```
for (stat_key, stat_value), (stats_model) in zip(stats_dict.items(), stats_models_to_update):
```

If we use it like that we still can not confirm whether the order of `stats_dict` matches for `stats_models_to_update`?